repos/spack_repo/builtin/packages/py_docutils/package.py

```diff
@@ -34,10 +35,14 @@ class PyDocutils(PythonPackage):
     version("0.12", sha256="c7db717810ab6965f66c8cf0398a98c9d8df982da39b4cd7f162911eb89596fa")
 
     depends_on("python@3.7:", when="@0.19:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
-
-    # Uses 2to3
-    depends_on("py-setuptools@:57", when="@:0.15", type="build")
+    depends_on("python@3.9:", when="@0.21:", type=("build", "run"))
+
+    with when("@0.21:"):
+        depends_on("py-flit-core@3.11:3", type="build")
```

Comment:
According to the packages on pypi and also the even older history of `pyproject.toml` (https://github.com/docutils/docutils/blame/148eac39f852f46981f02a796e8395f76cf353d9/docutils/pyproject.toml) `py-docutils@0.21:` is fine with `py-flit-core@3.4` as well.

```suggestion
        depends_on("py-flit-core@3.4:3", type="build")
```

Reply:
Does this work? https://github.com/docutils/docutils/commit/afec5e5801c37feac1356e84a96f321003b09cd5 indicates that this may have been needed to support the `license-files` field.

Reply:
For me it works with a quick test (builds fine with `py-flit-core@3.4` and runs with `python -c "import docutils"`).

If you think that there is really a problem with it, I am also fine with the newer version constraint if you add a comment  to why that differs from the `pyproject.toml`

Reply:
Sorry, I see it now. I thought the commit upgrading to py-flit-core@3.11 was in 0.21 already.